File: frontend/src/types/reviewType.ts
export interface ReviewType {
  id: number;
  name: string;
  status: string;
  project_category: string | null;
  sub_category: string | null;
  created_by: number | null;
  created_at: string | null;
  updated_at: string | null;
  template_count?: number;
  active_template_count?: number;
  expired_template_count?: number;
}

File: backend/app/schemas/review_type.py
from pydantic import BaseModel, Field
from typing import Optional


class ReviewTypeCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    project_category: Optional[str] = None
    sub_category: Optional[str] = None


class ReviewTypeUpdate(BaseModel):
    name: Optional[str] = None
    project_category: Optional[str] = None
    sub_category: Optional[str] = None


class ReviewTypeStatusUpdate(BaseModel):
    status: str  # active or inactive


class ReviewTypeResponse(BaseModel):
    id: int
    name: str
    status: str
    project_category: Optional[str] = None
    sub_category: Optional[str] = None
    created_by: Optional[int] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None
    template_count: Optional[int] = 0
